mlflow/pyfunc/__init__.py

```diff
@@ -2098,6 +2098,21 @@ def spark_udf(
     # can't access NFS, so it needs to use `spark.addArtifact`.
     use_dbconnect_artifact = is_dbconnect_mode and not is_in_databricks_shared_cluster_runtime()
 
+    if use_dbconnect_artifact:
+        udf_sandbox_info = get_dbconnect_udf_sandbox_info(spark)
+        if Version(udf_sandbox_info.mlflow_version) < Version("2.19.0"):
+            raise MlflowException(
+                "Using 'mlflow.pyfunc.spark_udf' in Databricks Serverless or in remote "
+                "Databricks Connect requires UDF sandbox image installed with MLflow "
+                "of version >= 2.19."
+            )
+        # `udf_sandbox_info.runtime_version` format is like '<major_version>.<minor_version>'.
+        if Version(udf_sandbox_info.runtime_version) < Version("16.0"):
```

Comment:
```suggestion
        if Version(udf_sandbox_info.runtime_version).major < 16:
```

to avoid unnecessary parsing.